controllers/aws_config_loader.go

```diff
@@ -42,15 +41,24 @@ func NewAwsConfigLoaderForEnv(env venv.Env) AwsConfigLoader {
 	}
 }
 
+// CreateNewAWSSessionFromConfig returns an AWS session using AWS Config
+func CreateNewAWSSessionFromConfig(cfg aws.Config) *awssession.Session {
+	sess, _ := awssession.NewSessionWithOptions(
+		awssession.Options{
+			SharedConfigState: awssession.SharedConfigEnable,
+			Config:            cfg,
+		})
+	return sess
+}
+
 // LoadAwsConfigWithOverrides loads default AWS config and apply overrides, like setting the region and using a custom SageMaker endpoint.
 // If specified, jobSpecificEndpointOverride always overrides the endpoint. Otherwise, the environment
 // variable specified by DefaultSageMakerEndpointEnvKey overrides the endpoint if it is set.
 func (l AwsConfigLoader) LoadAwsConfigWithOverrides(regionOverride string, jobSpecificEndpointOverride *string) (aws.Config, error) {
 	var config aws.Config
-	var err error
 
-	if config, err = external.LoadDefaultAWSConfig(external.WithRegion(regionOverride)); err != nil {
-		return aws.Config{}, err
+	if regionOverride != "" {
```

Comment:
Comparison to empty string is an antipattern for checking if something should exist. Perhaps changing `regionOverride` to `*string` and then checking `regionOverride != nil` would be cleaner.

Reply:
The default value for `string` variable is "" so it seems a reasonable check to me.

Any string reason why we should not do this ?

Reply:
We pass `jobSpecificEndpointOverride` as a pointer but `regionOverride` as a value. Typically if we mean "no value" then, in Go at least, we want to pass a `nil` pointer - rather than a "". This is the equivalent of passing `None` instead of `""` in Python.

Reply:
still not sold on empty string comparison 
but for consistency changing `regionOverride` to pointer and aws.Config expects it to be pointer anyway